substrate/frame/support/src/traits/hooks.rs

```diff
@@ -227,6 +227,26 @@ pub trait OnRuntimeUpgrade {
 	}
 }
 
+/// See [`Hooks::on_runtime_upgrade`].
+pub trait UncheckedOnRuntimeUpgrade {
+	/// See [`Hooks::on_runtime_upgrade`].
+	fn unchecked_on_runtime_upgrade() -> Weight {
```

Comment:
I *slightly* prefer this just being 

```suggestion
	fn on_runtime_upgrade() -> Weight {
```

Since it's already clear from the trait name that it's unchecked.